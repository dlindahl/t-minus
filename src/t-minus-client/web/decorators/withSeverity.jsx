import Colors from '../../shared/constants/Colors';
import { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import TimerSeverity from '../../shared/actionTypes/TimerSeverityTypes';

const theme = {
  [TimerSeverity.TIMER_SEVERITY_CALM]: {
    primaryColor: Colors.MidnightBlue,
    secondaryColor: Colors.Silver,
    tertiaryColor: Colors.Concrete
  },
  [TimerSeverity.TIMER_SEVERITY_WARN]: {
    primaryColor: Colors.MidnightBlue,
    secondaryColor: Colors.SunFlower,
    tertiaryColor: Colors.Orange
  },
  [TimerSeverity.TIMER_SEVERITY_DANGER]: {
    primaryColor: Colors.MidnightBlue,
    secondaryColor: Colors.Alizarin,
    tertiaryColor: Colors.Pomegranate
  },
  [TimerSeverity.TIMER_SEVERITY_CRITICAL]: {
    primaryColor: Colors.Clouds,
    secondaryColor: Colors.Pomegranate,
    tertiaryColorcolor: Colors.Alizarin
  }
};

function getState(state) {
  return {
    severity: state.display.severity
  };
}

export default function withSeverity(WrappedComponent) {
  @connect(getState)
  class SeverityTheme extends Component {
    render() {
      return <WrappedComponent {...this.props} {...theme[this.props.severity]}/>
    }
  }
  SeverityTheme.propTypes = {
    severity: PropTypes.string
  };
  SeverityTheme.defaultProps = {
    severity: TimerSeverity.TIMER_SEVERITY_CALM
  };
  SeverityTheme.WrappedComponent = WrappedComponent;
  return SeverityTheme;
}
