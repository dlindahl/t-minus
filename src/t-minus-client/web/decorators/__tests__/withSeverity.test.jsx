import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer'
import TimerSeverity from '../../../shared/actionTypes/TimerSeverityTypes'
import withSeverity from '../withSeverity'

const SeverityComponent = withSeverity(({ primaryColor, secondaryColor, tertiaryColor }) => (
  <div primaryColor={primaryColor} secondaryColor={secondaryColor} tertiaryColor={tertiaryColor}/>
))
const states = {
  [TimerSeverity.TIMER_SEVERITY_CALM]: {
    display: {
      severity: TimerSeverity.TIMER_SEVERITY_CALM
    }
  },
  [TimerSeverity.TIMER_SEVERITY_WARN]: {
    display: {
      severity: TimerSeverity.TIMER_SEVERITY_WARN
    }
  },
  [TimerSeverity.TIMER_SEVERITY_DANGER]: {
    display: {
      severity: TimerSeverity.TIMER_SEVERITY_DANGER
    }
  },
  [TimerSeverity.TIMER_SEVERITY_CRITICAL]: {
    display: {
      severity: TimerSeverity.TIMER_SEVERITY_CRITICAL
    }
  }
}
const store = configureStore()

describe('withSeverity', () => {
  it('passes in appropriate theme colors when the severity state is TIMER_SEVERITY_CALM', () => {
    const component = renderer.create(
      <SeverityComponent store={store(states[TimerSeverity.TIMER_SEVERITY_CALM])}/>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('passes in appropriate theme colors when the severity state is TIMER_SEVERITY_WARN', () => {
    const component = renderer.create(
      <SeverityComponent store={store(states[TimerSeverity.TIMER_SEVERITY_WARN])}/>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('passes in appropriate theme colors when the severity state is TIMER_SEVERITY_DANGER', () => {
    const component = renderer.create(
      <SeverityComponent store={store(states[TimerSeverity.TIMER_SEVERITY_DANGER])}/>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('passes in appropriate theme colors when the severity state is TIMER_SEVERITY_CRITICAL', () => {
    const component = renderer.create(
      <SeverityComponent store={store(states[TimerSeverity.TIMER_SEVERITY_CRITICAL])}/>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
