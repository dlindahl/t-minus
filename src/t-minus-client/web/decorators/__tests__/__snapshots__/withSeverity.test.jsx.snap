exports[`withSeverity passes in appropriate theme colors when the severity state is TIMER_SEVERITY_CALM 1`] = `
<div
  primaryColor="#2c3e50"
  secondaryColor="#bdc3c7"
  tertiaryColor="#95a5a6" />
`;

exports[`withSeverity passes in appropriate theme colors when the severity state is TIMER_SEVERITY_CRITICAL 1`] = `
<div
  primaryColor="#ecf0f1"
  secondaryColor="#c0392b"
  tertiaryColor={undefined} />
`;

exports[`withSeverity passes in appropriate theme colors when the severity state is TIMER_SEVERITY_DANGER 1`] = `
<div
  primaryColor="#2c3e50"
  secondaryColor="#e74c3c"
  tertiaryColor="#c0392b" />
`;

exports[`withSeverity passes in appropriate theme colors when the severity state is TIMER_SEVERITY_WARN 1`] = `
<div
  primaryColor="#2c3e50"
  secondaryColor="#f1c40f"
  tertiaryColor="#f39c12" />
`;
